code/modules/roguetown/roguemachine/merchant.dm

```diff
@@ -135,6 +136,28 @@
 	set_light(1, 1, "#1b7bf1")
 	add_overlay(mutable_appearance(icon, "vendor-merch"))
 
+/obj/structure/roguemachine/merchantvend/attack_right(mob/user)
+	if(user.mind.assigned_role == "Shophand")
+		if(hidden_key)		
+			for(var/mob/living/carbon/human/boss in GLOB.human_list)		
+				if(boss.mind)
+					if(boss.mind.assigned_role == "Merchant")
```

Comment:
I hate this loop, I would really hope there's a list for this but I'm not going to ask you to find it.